contrib/golang/filters/http/source/go/pkg/http/type.go

```diff
@@ -81,53 +77,49 @@ func (h *httpHeaderMap) Values(key string) []string {
 	return value
 }
 
-func (h *httpHeaderMap) Set(key, value string) {
+func (h *requestOrResponseHeaderMapImpl) Set(key, value string) {
 	// Get all header values first before setting a value, since the set operation may not take affects immediately
 	// when it's invoked in a Go thread, instead, it will post a callback to run in the envoy worker thread.
 	// Otherwise, we may get outdated values in a following Get call.
-	if h.headers == nil && !h.isTrailer {
+	if h.headers == nil {
 		h.headers = cAPI.HttpCopyHeaders(unsafe.Pointer(h.request.req), h.headerNum, h.headerBytes)
 	}
 	if h.headers != nil {
 		h.headers[key] = []string{value}
 	}
-	if h.isTrailer {
-		cAPI.HttpSetTrailer(unsafe.Pointer(h.request.req), &key, &value)
-	} else {
-		cAPI.HttpSetHeader(unsafe.Pointer(h.request.req), &key, &value, false)
-	}
+
+	cAPI.HttpSetHeader(unsafe.Pointer(h.request.req), &key, &value, false)
 }
 
-func (h *httpHeaderMap) Add(key, value string) {
+func (h *requestOrResponseHeaderMapImpl) Add(key, value string) {
+	if h.headers == nil {
```

Comment:
nit: So many identical checks for the header, so we can try to separate it to the header initialization phase. same elseeverywhere.